File: src/main/java/com/jacuum/leaderboard/LeaderboardEntry.java
package com.jacuum.leaderboard;

import java.util.List;

public record LeaderboardEntry(
    String id,
    String username,
    String avatar,
    String mapHash,
    String mapSize,
    String algoName,
    int iterationsUsed,
    int iterationsAvailable,
    int score,
    String completedAt,
    List<TraceEvent> trace
) {}

File: src/main/kotlin/com/jacuum/leaderboard/LeaderboardEntry.kt
package com.jacuum.leaderboard

data class LeaderboardEntry(
    val id: String,
    val username: String,
    val avatar: String,
    val mapHash: String,
    val mapSize: String,
    val algoName: String,
    val iterationsUsed: Int,
    val iterationsAvailable: Int,
    val score: Int,
    val completedAt: String,
    val trace: List<TraceEvent>
)
